### DISCUSSION TITLE
Can only scroll up one page using SHFT+PgUp/PgDn

### DISCUSSION THREAD

**Discussion by mmmmmmmkay** [lxqt/lxqt#2109] [Q&A]
I added a new user in Kali Linux and for some reason when I try to scroll up using the keyboard shortcut SHFT+PgUp/PgDn it only scrolls up one page and no further.

If I use the mouse wheel, it works perfectly fine and I can scroll all the way up as usual.

Any suggestions on how I can fix this?

Thanks!

**Comment by tsujan** [comment]
In which app?

**Reply by mmmmmmmkay** [reply]
Just in terminal console emulator, no app or command in particular. For example, if I do cat file.txt and then want to scroll up to view beginning I can't within QTerminal. Not sure why, I've been able to scroll up and down using SHFT+PGDN/PGUP but now I can't. This doesn't happen in other kali instances that I have... wondering if there's something I need to set somewhere to fix this

**Reply by tsujan** [reply]
I can't reproduce the issue in QTerminal; SHFT+PgUp/PgDn make QTerminal's view scroll up/down. I have the latest git qtermwidget+qterminal.

Rename `~/.config/qterminal.org/qterminal.ini` and try again. If the problem persists, it may be related to the packages of qtermwidget and qterminal or the version you have.

**Reply by tsujan** [reply]
Also check whether your PGDN/PGUP keys work fine in other apps.

**Reply by mmmmmmmkay** [reply]
Hmmm this happened after I did an apt-get upgrade. I just read a post that the linux kernel no longer supports scrollback -- do you think this is the cause? https://www.linuxquestions.org/questions/slackware-14/the-latest-kernel-release-4175597503/page52.html#post6167719

I found that from this post: https://unix.stackexchange.com/questions/325542/linux-console-shift-pgup-not-working-anymore

**Reply by tsujan** [reply]
Interesting info!

I only use LTS kernels. It's 5.10.75 now and I don't see this problem with it, while, according to your second link, the change should have been made to 5.8.10.

**Reply by tsujan** [reply]
I checked XTerm too. Like in QTerminal, SHFT+PgUp/PgDn works with it here.

**Reply by mmmmmmmkay** [reply]
So I just compared the qterminal.ini file from the instance that is working vs. the qterminal.ini file that is not working.

The version that is working (0.14.1) has the following line:
HistoryLimited=true

The version that is not working (0.16.1) has the following line:
HistoryLimited=false

When I try to change the version that is not working to True (HistoryLimited=true) and save the file, for some reason the next time I launch qterminal it just reverts back to false.

Do you know why I can't manually set this parameter to true in qterminal.ini file?

Also I noticed in qterminal version 0.14.1, it does not have the parameter "HandleHistory" in the .ini file but in version 0.16.1 it has the following with no value set:
HandleHistory=

Any suggestions? Thanks again for all your help on this!

**Reply by tsujan** [reply]
> Do you know why I can't manually set this parameter to true in qterminal.ini file?

Close all instances of QTerminal before manual editing. If you have a QTerminal instance running, it'll overwrite the config file.

**Reply by mmmmmmmkay** [reply]
So I figured out the problem! It turns out the culprit was Preferences > Behavior > Action after paste

It was set to Move end but after setting it to No move, I can once again scroll up and down using the SHFT+PgUp/PgDn keys 

Sweet sweet scroll up/down... =)

**Reply by tsujan** [reply]
Please also note that 0.16 is outdated.

**Reply by mmmmmmmkay** [reply]
I used apt-get update && apt-get install qterminal ... shouldn't that grab the latest version? Sorry I've only been using linux for a few months now :S

**Reply by tsujan** [reply]
> shouldn't that grab the latest version?

Not necessarily, but it doesn't mean that you should do something else. Just wait for new packages in your distro.

> I've only been using linux for a few months now 

I explain it briefly.

A program is developed by its developer(s), like LXQt and its components on GitHub. Then, each distro (like Debian, Ubuntu, Fedora...) may have a package maintainer, who makes installable packages out of that program for that distro. Whether those packages are up-to-date or not depends on the kind of distro (rolling vs. non-rolling or LTS) or if it has active maintainers for that particular program.

Therefore, LXQt is not responsible for distro packages.

**Reply by mmmmmmmkay** [reply]
Ah okay that makes sense. Well I appreciate all your help, and other than this one issue I had, QTerminal has been awesome and has not failed me!! 

Thank you sir!
